src/Microsoft.DotNet.Interactive.Documents/InteractiveDocument.cs

```diff
@@ -241,13 +241,68 @@ internal static void MergeKernelInfos(KernelInfoCollection destination, KernelIn
     {
         if (metadata is not null)
         {
-            if (metadata.TryGetValue("kernelInfo", out var kernelInfoObj) &&
-                kernelInfoObj is JsonElement kernelInfoJson && kernelInfoJson.Deserialize<KernelInfoCollection>(ParserServerSerializer.JsonSerializerOptions) is
-                    { } kernelInfoDeserialized)
+            if (metadata.TryGetValue("kernelInfo", out var kernelInfoObj) )
             {
-                kernelInfo = kernelInfoDeserialized;
-                return true;
+                if (kernelInfoObj is JsonElement kernelInfoJson &&
+                    kernelInfoJson.Deserialize<KernelInfoCollection>(ParserServerSerializer.JsonSerializerOptions) is
+                        { } kernelInfoDeserialized)
+                {
+                    kernelInfo = kernelInfoDeserialized;
+                    return true;
+                }
+
+                // todo: the kernelInfo should not deserialize as a dictionary
+                if (kernelInfoObj is Dictionary<string,object> kernelInfoAsDictionary)
```

Comment:
Is there a test for the issue this is intended to resolve?

Reply:
Just added a test that uses serialize and parse request. Serialize touches the new code. The document used in test comes from a vscode serialize request.